### DISCUSSION TITLE
Jsonnet support?

### DISCUSSION THREAD

**Discussion by 7840vz** [perses/perses#2220] [General]
Do you have any plans for jsonnet SDK support? jsonnet is quite popular for building Grafana dashboards and their latest library is generated from cuelang schema.

https://github.com/grafana/grafonnet

**Comment by Nexucis** [comment]
From my perspective, what we are providing (aka Dashboard As Code SDK in Golang and in Cuelang) is much better than what you can do with Jsonnet.

Perhaps @AntoineThebaud @celian-garcia have a different opinion.

**Comment by AntoineThebaud** [comment]
Well regarding supporting SDKs for other (programming or templating) languages, our current statement is pretty much: contributions are welcome 😄 . We have so much topics to work on at the moment that developping & maintaining ourselves other SDKs than CUE and Go is not at all in the plans. Still, if other people would be willing to work on this, we'd be happy to help & review the contributions.

**Comment by 7840vz** [comment]
Just saw a ticket here in Thanos:
https://github.com/thanos-io/thanos/issues/7702

I wonder if generating Jsonnet lib from cuelang definitions (just like [grafonnet](https://github.com/grafana/grafonnet) does from grafana cuelang defintions) would help accepting perses as part of  monitoring-mixins format?

With extended [monitoring-mixins](https://monitoring.mixins.dev/) schema like:

```
{
  grafanaDashboards+:: {},
  prometheusRules+:: {},
  prometheusAlerts+: {},
  persesDashboards+: {},
}
```

P.S. btw, perhaps it is best adding support to persesDashboards+:{} to https://monitoring.mixins.dev/  repo, not Thanos repo?

**Reply by Nexucis** [reply]
I think if community want to generate jsonnet based on the cue files we are providing, we are not blocking that.

On Perses side, that's not something we are looking for at this moment and I would prefer to encourage people to use the SDKs we are providing.

Besides they can still generate grafana dashboard with Jsonnet and then run the migration script.

**Reply by Nexucis** [reply]
About the repo ` https://monitoring.mixins.dev/ ` it looks like it is just a website that references the mixins. My assumption is based on this file: https://github.com/monitoring-mixins/website/blob/master/mixins.json

**Reply by Nexucis** [reply]
so if we want to add perses dashboard for Thanos or Prometheus it should go in their respective repository.

**Comment by Nexucis** [comment]
In the effort to provide default dashboard in Perses, this repository has been created: https://github.com/perses/community-dashboards

It contains the default monitoring dashboard for Prometheus / Thanos / Kubernetes. As Jsonnet is the default format for mixins and lackely the different project won't support anything else, this repo is generating the various dashboards in the Jsonnet format.

Perses itself won't support Jsonnet, but tools that integrate it will.
